nni/algorithms/compression/pytorch/auto_compress/auto_compress_engine.py

```diff
@@ -0,0 +1,150 @@
+
+        config_list = []
+        for key, config in config_dict.items():
+            op_types, op_names = key
+            op_types = op_types.split(':') if op_types else []
+            op_names = op_names.split(':') if op_names else []
+            if op_types:
+                config['op_types'] = op_types
+            if op_names:
+                config['op_names'] = op_names
+            if 'op_types' in config or 'op_names' in config:
+                config_list.append(config)
+
+        return config_list
+
+    @classmethod
+    def __convert_quantizer_config_list(cls, converted_config_dict: dict) -> list:
+        config_dict = {}
+        for key, value in converted_config_dict.items():
+            _, quant_types, op_types, op_names, var_name = key.split('::')
+            config_dict.setdefault((quant_types, op_types, op_names), {})
+            config_dict[(quant_types, op_types, op_names)][var_name] = value
+
+        config_list = []
+        for key, config in config_dict.items():
+            quant_types, op_types, op_names = key
+            quant_types = quant_types.split(':')
+            op_types = op_types.split(':')
+            op_names = op_names.split(':')
+            if quant_types:
+                config['quant_types'] = quant_types
+            else:
+                continue
+            if op_types:
+                config['op_types'] = op_types
+            if op_names:
+                config['op_names'] = op_names
+            if 'op_types' in config or 'op_names' in config:
+                config_list.append(config)
+
+        return config_list
+
+    @classmethod
+    def _convert_config_list(cls, compressor_type: str, converted_config_dict: dict) -> list:
+        func_dict = {
+            'pruner': cls.__convert_pruner_config_list,
+            'quantizer': cls.__convert_quantizer_config_list
+        }
+        return func_dict[compressor_type](converted_config_dict)
+
+    @classmethod
+    def __compress_pruning_pipeline(cls, algorithm_name: str,
+                                    model: Module,
+                                    config_list: list,
+                                    evaluator: Callable[[Module], float],
+                                    optimizer: Optional[Optimizer],
+                                    trainer: Optional[Callable[[Module, Optimizer], None]],
+                                    finetune_trainer: Optional[Callable[[Module, Optimizer], None]],
+                                    **compressor_parameter_dict) -> Module:
+        # evaluator is for future use
+        pruner = PRUNER_DICT[algorithm_name](model, config_list, optimizer, **compressor_parameter_dict)
+        model = pruner.compress()
+        if trainer:
+            trainer(model)
+        if finetune_trainer:
+            finetune_trainer(model)
+        return model
+
+    @classmethod
+    def __compress_quantization_pipeline(cls, algorithm_name: str,
```

Comment:
the two functions can be merged together

Reply:
it has a little different, pls see the new update, I think it’s clearer to implement separately.